### DISCUSSION TITLE
Default Hugo template appears broken

### DISCUSSION THREAD

**Discussion by leblowl** [vercel/community#6472] [Help] [answered]
### Summary

Hi, perhaps I'm using things incorrectly, but I'm new to Vercel and I just started a default Hugo template and first received this error when deploying:

```
Error: Found invalid Node.js Version: "18.x". Please set "engines": { "node": "20.x" } in your `package.json` file to use Node.js 20.
```

I updated the Node version in package.json and then received this error:

```
sh: line 1: hugo: command not found
Error: Command "hugo --gc" exited with 127
```

I was expecting everything in the build environment to be available and ready to go.

Thanks

### Steps to Reproduce

Simply create a new Hugo template and try to deploy it in the Vercel web app

**Comment by swarnava** [comment]
Can you share the link of the template?

**Reply by leblowl** [reply]
Here is the repository: https://github.com/leblowl/hugo

Is there anything else I can provide?

**Comment by versecafe** [comment]
@leblowl what is your node version on Vercel

<img width="1380" alt="image" src="https://github.com/vercel/community/assets/147033096/dcdd310f-a8d7-4cb7-8b1e-101ed391ffc5">

**Comment by TooTallNate** [comment, marked as answer]
The [Build Image](https://vercel.com/docs/deployments/build-image) that is used when Node Version `20.x` is selected does not contain the `hugo` executable. Instead, you should select `18.x` to use the build image which does contain `hugo`.

**Comment by leblowl** [comment]
Well I'm not really sure what's going on. I joined Vercel, chose the Hugo template in Vercel and it didn't build (nothing custom), so just informing ya'll as there might be a bug here.

When I looked just now, the Node version in Vercel was set to 18. And yet I received that error above 2 days ago:
```
Error: Found invalid Node.js Version: "18.x". Please set "engines": { "node": "20.x" } in your `package.json` file to use Node.js 20.
```

Just now, I downgraded the Node version in package.json (from 20 to 18), and switched the project Node version in Vercel from 18 to 20 and it built. I then downgraded the project Node version in Vercel from 20 to 18 and it still built.

I just created a new Hugo template in Vercel and the project Node version is set to Node 20 (package.json version set to 18) and this time it built successfully without any modification (https://github.com/leblowl/hugo2/tree/main).

Your docs say the Node 20 build image doesn't contain Go, but it looks like it does to me (if I'm understanding correctly that the project Node version controls the build image... I don't see any logs in the actual deployment that tells me what image it's using).

Anyways, I don't really need a specific answer or resolution out of this, just providing this information in case it's helpful to ya'll.

**Reply by swarnava** [reply]
Go support was added with latest Node 20 image 9 hours ago. It should be working now.

**Reply by jckinjo** [reply]
No, the hugo is still executable when Node `20.x` is selected 

```
sh: line 1: hugo: command not found
Error: Command "hugo --gc" exited with 127
```
